src/Storages/ObjectStorageQueue/ObjectStorageQueueIFileMetadata.cpp

```diff
@@ -259,15 +259,33 @@ bool ObjectStorageQueueIFileMetadata::trySetProcessing()
     return success;
 }
 
-ObjectStorageQueueIFileMetadata::SetProcessingResponseIndexes
+std::optional<ObjectStorageQueueIFileMetadata::SetProcessingResponseIndexes>
 ObjectStorageQueueIFileMetadata::prepareSetProcessingRequests(Coordination::Requests & requests)
 {
-    [[maybe_unused]] auto state = file_status->state.load();
-    chassert (!(state == FileStatus::State::Processing
+    std::unique_lock processing_lock(file_status->processing_lock, std::defer_lock);
+    if (!processing_lock.try_lock())
+    {
+        /// This is possible in case on the same server
+        /// there are more than one S3(Azure)Queue table processing the same keeper path.
+        LOG_TEST(log, "File {} is being processed on this server by another table on this server", path);
+        return std::nullopt;
+    }
+
+    auto state = file_status->state.load();
```

Comment:
we can now have multiple workers taking the same file and trying to set it as processing
chance of having a collision is high in that case, so multi request failure rate could be high

if a node is part of hash ring, node should probably be responsible for assigning files to specific local worker

Reply:
As clarified above, it would only happen in case of separate s3queue tables on the same server, but in this case they have the same metadata instance, but registering as active processor is made on metadata instance level, which would be shared for all server-local s3queue tables.